lib/src/working_copy.rs

```diff
@@ -374,6 +376,30 @@ pub enum ResetError {
     InternalBackendError(#[from] BackendError),
 }
 
+#[derive(Debug, Error)]
+pub enum TreeStateError {
+    #[error("Reading tree state from {}: {source}", .path.display())]
+    ReadTreeState {
```

Comment:
nit: surprisingly, this error supports `"{path}"`

Reply:
TIL! Thanks, much more convenient.